### PR TITLE
2.x: add ParallelTransformer interface, params-validation

### PR DESCRIPTION
**Pull request by akarnokd**
This PR adds the `ParallelTransformer` interface to match the other `XTransformer` interfaces, adds `@NonNull` annotations to the `ParallelFlowable` operators and adds the `ParallelFlowable` class to the parameter validator test set.

### CODE REVIEW

File: src/main/java/io/reactivex/parallel/ParallelFlowable.java [hunk 1]
```diff
@@ -629,8 +629,8 @@ protected final boolean validate(Subscriber<?>[] subscribers) {
      * @return the ParallelFlowable returned by the function
      */
     @CheckReturnValue
-    public final <U> ParallelFlowable<U> compose(Function<? super ParallelFlowable<T>, ParallelFlowable<U>> composer) {
-        return RxJavaPlugins.onAssembly(to(composer));
+    public final <U> ParallelFlowable<U> compose(@NonNull ParallelTransformer<T, U> composer) {
+        return RxJavaPlugins.onAssembly(composer.apply(this));
```
Comment (vanniktech) [inline src/main/java/io/reactivex/parallel/ParallelFlowable.java]: Missing check here for composer being null?

File: src/main/java/io/reactivex/parallel/ParallelFlowable.java [hunk 2]
```diff
@@ -611,7 +611,7 @@ protected final boolean validate(Subscriber<?>[] subscribers) {
      * @return the value returned by the converter function
      */
     @CheckReturnValue
-    public final <U> U to(Function<? super ParallelFlowable<T>, U> converter) {
+    public final <U> U to(@NonNull Function<? super ParallelFlowable<T>, U> converter) {
         try {
             return converter.apply(this);
```
Comment (vanniktech) [inline src/main/java/io/reactivex/parallel/ParallelFlowable.java]: Missing check here for converter being null

**Review by vanniktech — APPROVED** [review]

File: src/main/java/io/reactivex/parallel/ParallelFlowable.java (hunk 2 above)
Comment (akarnokd) [reply src/main/java/io/reactivex/parallel/ParallelFlowable.java]: Dereference will throw NPE automatically.

Comment (vanniktech) [reply src/main/java/io/reactivex/parallel/ParallelFlowable.java]: I'd still check it in order to be consistent and for the better error message knowing what exactly is null.

Comment (akarnokd) [reply src/main/java/io/reactivex/parallel/ParallelFlowable.java]: Updated this and all other methods.

**Review by vanniktech — APPROVED** [review]
Awesome 👍

**Pull request merged**